tests/Unit/Helpers/Domain/DomainTestHelpers.cpp

```diff
@@ -382,30 +384,75 @@ void test_domain_construction(
+          CHECK(block.external_boundaries() == expected_external_boundaries[i]);
+          dispatch_check_if_maps_are_equal(block, *expected_maps[i], time,
+                                           functions_of_time);
+          if (std::is_same<TargetFrameGridOrInertial, Frame::Grid>::value) {
+            if (expected_grid_to_inertial_maps.size() != blocks.size()) {
+              ERROR("Need at least one grid to inertial map for each block ("
+                    << blocks.size() << " blocks in total) but received "
+                    << expected_grid_to_inertial_maps.size() << " instead.");
+            }
+            dispatch_check_if_maps_are_equal(block,
+                                             *expected_grid_to_inertial_maps[i],
+                                             time, functions_of_time);
+          }
+        }
+        if (not expected_boundary_conditions.empty()) {
+          for (size_t block_id = 0;
+               block_id < expected_boundary_conditions.size(); ++block_id) {
+            CAPTURE(block_id);
+            REQUIRE(expected_boundary_conditions[block_id].size() ==
+                    expected_external_boundaries[block_id].size());
+            REQUIRE(expected_boundary_conditions[block_id].size() ==
+                    blocks[block_id].external_boundary_conditions().size());
+            const auto& boundary_conditions_in_block =
+                blocks[block_id].external_boundary_conditions();
+            for (const auto& [direction, expected_boundary_condition_ptr] :
+                 expected_boundary_conditions[block_id]) {
+              CAPTURE(direction);
+              CAPTURE(boundary_conditions_in_block.size());
+              REQUIRE(boundary_conditions_in_block.count(direction) == 1);
+              REQUIRE(boundary_conditions_in_block.at(direction) != nullptr);
+              const auto& boundary_condition_in_direction =
+                  dynamic_cast<const TestHelpers::domain::BoundaryConditions::
+                                   TestBoundaryCondition<VolumeDim>&>(
+                      *boundary_conditions_in_block.at(direction));
+              const auto& expected_boundary_condition =
+                  dynamic_cast<const TestHelpers::domain::BoundaryConditions::
+                                   TestBoundaryCondition<VolumeDim>&>(
+                      *expected_boundary_condition_ptr);
+              CHECK(expected_boundary_condition.direction() ==
+                    boundary_condition_in_direction.direction());
+              CHECK(expected_boundary_condition.block_id() ==
+                    boundary_condition_in_direction.block_id());
+            }
+          }
+        }
+      };
+  perform_checks(domain);
   domain::creators::register_derived_with_charm();
+  TestHelpers::domain::BoundaryConditions::register_derived_with_charm();
+  INFO("Post-serialization");
   test_serialization(domain);
+  perform_checks(serialize_and_deserialize(domain));
```

Comment:
(optional) I find the catch2 output more useful when the `INFO`s are scoped:

```cpp
{
  INFO("Serialization");
  test_serialization(domain);
  perform_checks(serialize_and_deserialize(domain));
}
```

Then the "Serialization" only prints when the error occurred in the scope.